### DISCUSSION TITLE
Recharts scaling of graph in x axis

### DISCUSSION THREAD

**Discussion by mongchanghsi** [recharts/recharts#3605] [Q&A]
I am trying to build a line chart, that is Price (X-Axis) vs Quantity Available (Y-Axis). Say I have the following data points
```
 const datapoints = {
        price: 0.0001,
        quantity: 58,
      },
      {
        price: 0.5,
        quantity: 180,
      },
      {
        price: 1.2,
        quantity: 205,
      },
      {
        price: 5.1,
        quantity: 305,
      },
      {
        price: 20,
        quantity: 315,
      },
      {
        price: 30,
        quantity: 342,
      },
      {
        price: 500,
        quantity: 362,
      },
      {
        price: 999999,
        quantity: 382,
},
```

Given how the upper limit of the price is large compared to the lower limit, the graph will look essentially almost like a horizontal straight line.

![image](https://github.com/recharts/recharts/assets/32842582/61f69d92-2055-40a6-ab3b-9852a6e0063b)

Is it possible to scale it in a way such that the lower price take up a larger area as compared the higher price. So it will look more like a concave curve.

I noticed that there is a `scale` attribute under XAxis, I am not sure whether is it that, so far tried all the available default options is doesn't provide the result that I am looking for. Also when I tried `scale="log`, it has the following error. Not exactly sure what is the issue, but I had set my `width="100%"` under ResponsiveContainer
![image](https://github.com/recharts/recharts/assets/32842582/57eea644-b4fd-47b3-913f-510f82fa4361)
![image](https://github.com/recharts/recharts/assets/32842582/07ef2e31-603b-466a-8c79-254538be320c)

I am extremely new to Recharts or data visualisation as a whole, appreciate everyone's input!

**Comment by ckifer** [comment]
`scaleSymlog` from d3 works it seems although there is a type issue. Will have to look into why things are breaking - thanks!

https://codesandbox.io/s/simple-area-chart-forked-c3cyl0?file=/src/App.tsx

**Reply by mongchanghsi** [reply]
Thank you for the response, will try it out in abit! I am using v2.1.12 if it helps.

**Reply by mongchanghsi** [reply]
If I were to use
`[
  {
    price: 0.009,
    quantity: 15248,
  },
  {
    price: 0.00919,
    quantity: 18338,
  },
  {
    price: 0.0092,
    quantity: 21483,
  },
  {
    price: 0.0093,
    quantity: 25383,
  },
  {
    price: 0.0094,
    quantity: 43000,
  },
  {
    price: 0.0095,
    quantity: 54736,
  },
  {
    price: 0.01,
    quantity: 57568,
  },
  {
    price: 0.09,
    quantity: 63957,
  },
  {
    price: 0.1,
    quantity: 63960,
  },
  {
    price: 0.15,
    quantity: 103188,
  },
  {
    price: 0.2,
    quantity: 103275,
  },
  {
    price: 0.3,
    quantity: 103375,
  },
  {
    price: 0.2,
    quantity: 103376,
  },
  {
    price: 10,
    quantity: 103377,
  },
  {
    price: 22222,
    quantity: 103397,
  },
  {
    price: 44454,
    quantity: 103408,
  },
  {
    price: 90909,
    quantity: 103413,
  },
]`

![image](https://github.com/recharts/recharts/assets/32842582/aad133a6-8f09-4e90-a5d6-7696b060ef08)

I think I will have to adjust the domain and the range within the `scaleSymLog` but I don't quite get how am I suppose to adjust it. Tried doing
`scaleSymlog([datapoints[0]?.price, datapoints[datapoints.length - 1]?.price], [0, 100])`
